pkg/chains/signing.go

```diff
@@ -58,21 +62,45 @@ type TaskRunSigner struct {
 }
 
 // IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+func IsSigned(tr *v1beta1.TaskRun) (bool, string) {
 	signature, ok := tr.ObjectMeta.Annotations[ChainsAnnotation]
 	if !ok {
-		return false
+		return false, ""
 	}
-	return signature == "true"
+	return signature == "true", signature
+}
+
+// IsRetrying determines whether a TaskRun is retrying state.
+func IsRetrying(tr *v1beta1.TaskRun) (bool, string) {
```

Comment:
Instead of this, a function that just tells us if we should retry might be simpler to work in:

```go
// returns true if:
// the signed annotation isn't there, the failed annotation isn't there, and retries<3
func RetryAgain(tr *v1beta1.TaskRun) bool {}

Reply:
Changed it to RetryAgain.